core/runtime/src/main/java/io/quarkus/runtime/ExecutorRecorder.java

```diff
@@ -57,8 +58,13 @@ public void run() {
         if (threadPoolConfig.prefill) {
             underlying.prestartAllCoreThreads();
         }
-        current = underlying;
-        return underlying;
+        ScheduledExecutorService managed = underlying;
+        // In prod mode we don't want to allow the executor to be shut down by the application or other extensions
+        if (launchMode == LaunchMode.NORMAL) {
+            managed = new ManagedScheduledExecutorService(underlying);
+        }
+        current = managed;
+        return managed;
```

Comment:
Hm, isn't the Vert.x worker thread pool backed by this very `ExecutorService` in the prod mode? If so, then we could remove the vertx-specific stuff for the prod mode here: https://github.com/quarkusio/quarkus/blob/main/extensions/vertx/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java#L110

Reply:
I think so.

Reply:
Ok, let's try to simplify the code! Except that we don't have a test for this. I guess that we could at least try to run the reproducer from https://github.com/quarkusio/quarkus/issues/16833 manually?

Reply:
Ok, let me remove the draft to run the existing test suite. I can also reproduce with #43228.

Reply:
I suggest we allow the shutdown only on dev mode. It'd be better if tests were closer to prod.